airflow-core/tests/unit/api_fastapi/common/test_exceptions.py

```diff
@@ -282,3 +294,102 @@ def test_handle_multiple_columns_unique_constraint_error(
 
             assert response_detail == expected_detail
             assert "INSERT INTO dag_run" in actual_statement
+        assert exeinfo_response_error.value.detail == expected_exception.detail
+
+
+class TestDagErrorHandler:
+    dag_error_handler = DagErrorHandler()
+
+    def test_handle_deserialization_error_with_value_error(self):
+        error_message = "Missing Dag ID in serialized Dag"
+        deserialization_error = DeserializationError("test_dag_id")
+
+        value_error = ValueError(error_message)
+        deserialization_error.__cause__ = value_error
+
+        expected_exception = HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while trying to deserialize Dag: {deserialization_error}",
+        )
+
+        with pytest.raises(HTTPException) as exeinfo_response_error:
```

Comment:
Use `matches` param for `pytest.raises` instead to match the exception message.

Same for all tests actually, this will make the code shorted and remove the need for a manual extra assertion on the exception message.